tests/test_spider.py

```diff
@@ -92,6 +92,23 @@ def closed(self, reason):
                                        spider=spider, reason=None)
         self.assertTrue(spider.closed_called)
 
+    def test_idled_signal_call(self):
+        class TestSpider(self.spider_class):
+            idled_called = False
+
+            def idled(self):
+                self.idled_called = True
+
+        crawler = get_crawler()
+        spider = TestSpider.from_crawler(crawler, 'example.com')
+        crawler.signals.send_catch_log(signal=signals.spider_opened,
+                                       spider=spider)
+        crawler.signals.send_catch_log(signal=signals.spider_idle,
+                                       spider=spider)
+        crawler.signals.send_catch_log(signal=signals.spider_closed,
+                                       spider=spider, reason=None)
+        self.assertTrue(spider.idled_called)
+
```

Comment:
So of course I'm running into trouble with this.
Trying to do `self.assertRaises(DontCloseSpider, spider.idle(spider))` here, but I don't have an engine.

```py
            for request in cb_response:
>               spider.crawler.engine.crawl(request, spider)
E               AttributeError: 'NoneType' object has no attribute 'crawl'
```
setting the engine is not enough
`crawler.engine = crawler._create_engine()`
So it needs `mock.patch.object` ` crawler.engine.crawl` (how?) or something else entirely?

(Gah, this stuff is twisting my brain. i fear something's gonna snap.
Maybe I can bribe someone to misclick merge button while looking out the window?)